lib/ravenx/notification_behaviour.ex

```diff
@@ -3,5 +3,5 @@ defmodule Ravenx.NotificationBehaviour do
   Provides an interface for implementations of Ravenx notifications.
   """
 
-  @callback get_notifications_list(any) :: list
-end
\ No newline at end of file
+  @callback get_notifications_config(any) :: [atom: {atom, map} | {atom, map, map}]
```

Comment:
In the docs we define the data structure returned by `get_notifications_config` as:
```
1. Atom defining the notification ID.
2. A two or three element tuple containing:
  1. Atom defining which strategy should be used.
  2. Payload map with the data of the notification.
  3. (Optional) Options map for that strategy.
```

The typespec provides proper syntactic typing information, but maybe we could add some semantic information on top also. For this we could create some custom types whose names reveal their intention:

```elixir
@type notification_id :: atom
@type notification_strategy :: atom
@type notification_payload :: map
@type notification_options :: map
@type notification_config :: {notification_strategy, notification_payload} | {notification_strategy, notification_payload, notification_options}

@callback get_notifications_config(any) :: [notification_id: notification_config]
```